src/rotating_frame/rotating_frame.hpp

```diff
@@ -14,38 +14,294 @@
 #define ROTATING_FRAME_ROTATING_FRAME_HPP_
 
 // Parthenon includes
+#include <parthenon/driver.hpp>
 #include <parthenon/package.hpp>
 
 // Artemis includes
 #include "artemis.hpp"
+#include "derived/fill_derived.hpp"
 #include "geometry/geometry.hpp"
+#include "utils/artemis_utils.hpp"
+#include "utils/fluxes/reconstruction/reconstruction.hpp"
+#include "utils/integrators/artemis_integrator.hpp"
 
+using namespace parthenon::driver::prelude;
 using namespace parthenon::package::prelude;
+using ArtemisUtils::PLM;
+using ArtemisUtils::VI;
 
 namespace RotatingFrame {
 
+//----------------------------------------------------------------------------------------
+//! Declarations
 std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin);
 
 TaskStatus RotatingFrameForce(MeshData<Real> *md, const Real time, const Real dt);
 
+//----------------------------------------------------------------------------------------
+//! \fn std::array<Real, 3> RotatingFrame::ShearVelocity
+//! \brief Returns signed shear velocity
+template <Coordinates GEOM>
+KOKKOS_INLINE_FUNCTION Real ShearVelocity(const Real qshear, const Real omega,
+                                          const Real x1v) {
+  if constexpr (GEOM == Coordinates::cartesian) {
+    return -qshear * omega * x1v;
+  } else {
+    PARTHENON_FAIL("Shearing box currently only supports Cartesian geometries");
+  }
+}
```

Comment:
```suggestion
template <Coordinates GEOM>
KOKKOS_INLINE_FUNCTION 
std::array<Real,3> BackgroundVelocity(const std::array<Real,3> &x, const Real qshear, const Real omega) {
  std::array<Real,3> vb{0.0};
  if constexpr (GEOM == Coordinates::cartesian) {
    vb[1] = -qshear * omega * x[0];
  } else if constexpr( GEOM == Coordinates::cylindrical) {
    vb[1] = omega * x[0];
  } else if constexpr( GEOM == Coordinates::axisymmetric) {
    vb[2] = omega * x[0];
  } else if constexpr( GEOM == Coordinates::spherical) {
    vb[2] = omega * x[0]*std::sin(x[1]);
  }
  return vb;
}
```
This is more what I was thinking. Just return a zero vector if this isn't active and subtract / add zeros in places where it's awkward to have a switch with `do_shear` (if such places exist).

Reply:
This is overkill for now, since my edits in this MR only address the shearing box with `qshear != 0`.  Let's punt on this until we address more generally how to handle background flows.

Reply:
Let's compromise. I'm fine with removing the unused geometries but can we keep the return of the full 3D vector for the background velocity? The nbody package needs that full velocity in its sink method (even for shearing box) so it can add/subtract from the gas.

Reply:
Addressed in [5410d15](https://github.com/lanl/artemis/pull/68/commits/5410d158af596b360df47bda43566f281782e55a)